lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb

```diff
@@ -89,10 +89,44 @@ def custom_build_settings
             settings
           end
 
+          # @param [Generator] generator
+          #        the generator to use for generating the content.
+          #
+          # @param [Pathname] path
+          #        the pathname to save the content into.
+          #
+          # Saves the content the provided path unless the path exists and the contents are exactly the same.
+          #
+          # @return [Void]
+          #
+          def update_changed_file(generator, path)
+            if path.exist?
+              generator.save_as(support_files_temp_dir)
+              unless FileUtils.identical?(support_files_temp_dir, path)
+                generator.save_as(path)
+              end
+            else
+              generator.save_as(path)
+            end
+            clean_support_files_temp_dir if support_files_temp_dir.exist?
```

Comment:
It might be simpler but it'll be the same performance wise. Removing a directory doesn't have a syscall, so you will need to Recurse and remove each file individually in user space.